Update Nvidia Drivers to 381.22 and add suport for GT 1030

klondi opened the pull request:
In order to use LibreELEC with Nvidia GT 1030, drivers at version 381.22 or higher are needed.

Also, since card detection isn't working as intended because the HTML page used to generate the rules is missing the GT 1030 id I have added it manually.

This driver has been tested using escapade's build with the device fixes in the second commit on a computer with a GT 1030 where it worked flawlessly.

klondi commented:
Keep in mind that 381.22 belongs to the short term stable branch whilst 375.66 belongs to the long term stable branch before considering pushing this to prior releases.

MilhouseVH commented:
Hi @klondi. This crops up every now and again and we'd rather stick with the long-lived stable branch 375.xx for the time being. This branch is due an update real-soon-now which will most likely bring GT 1030 support in which case we can then bump with less risk.

Switching to an entirely new (and potentially less stable) driver branch for the sake of one new GPU isn't something we consider worthy of the risk, particularly as there are community builds that can (and do) fill in until such time as Nvidia adds official support in the long-lived branch.

This kind of change can and does result in flip-flopping back and forth between driver branches as new issues/bugs are discovered in the newer branch forcing us back to the long-lived branch, so sticking with the long-lived branch and waiting for the next update is the best option.

BTW the instructions in `xf86-video-nvidia/package.mk` tell you how to run the `make_nvidia_udev.py` script which will automatically generate the rules in future.

Closing (with thanks).

escalade commented:
@MilhouseVH 

FYI make_nvidia_udev.py simply pulls the ID's from that versions supportedchips.html file. The GT1030 has not been entered there even though it's supported.

MilhouseVH commented:
Ah yes, in that case it would need to be added manually.

I noticed the GT 1030 wasn't in the original 381.22 announcement on the developer forum:

https://devtalk.nvidia.com/default/topic/1007999

compared with the download page announcement:

http://www.nvidia.com/Download/driverResults.aspx/118524/en-us

Nor is the GT 1030 listed in the 381.22 [README/supportedchips](https://download.nvidia.com/XFree86/Linux-x86/381.22/README/supportedchips.html), so it looks like it's been added last minute or as an afterthought.

klondi commented:
Hey @MilhouseVH no worries and thanks for the reply and the amazing software. Let's hope the long term stable branch includes GT 1030 support.

As I said, I have been using it with @escalade's community build and 381.22 hence why I made this pull request as it was considered stable by Nvidia but I can understand why this won't be merged.

In theory with the GTX 1050 support being added there shouldn't be major changes to the drivers to support the GT 1030 as they share the same microarchitecture, but given the time between the releases maybe there are some differences (for example the GT1030 has a PCI-E 4.0x bus). That might be why they could add such support quite fast (supposing they just had to compile in the new card's hardware specs and PCI-IDs).

As for it not being listed I originally debugged and reported this on escalade's build see https://github.com/escalade/LibreELEC.tv/pull/31 I reran the script before doing this pull request to double check but it still didn't show up the GT 1030 pciid hence why I added it manually and moved it to a different commit.

Have a nice day!

MilhouseVH commented:
GT 1030 support is in 384.59 which is added by #1830 (master) and #1831 (libreelec-8.2).

This will be in my [test builds](https://forum.kodi.tv/showthread.php?tid=298462) from tonight, starting with build #0724.

384.xx is now the new long-lived branch.
